packages/plugin-core/src/telemetry.ts

```diff
@@ -1,87 +1,24 @@
-import { SegmentClient, TelemetryStatus } from "@dendronhq/common-server";
-import _ from "lodash";
-import * as vscode from "vscode";
-import { ExtensionProvider } from "./ExtensionProvider";
+import { ConfigUtils, DWorkspaceV2 } from "@dendronhq/common-all";
+import { SegmentClient } from "@dendronhq/common-server";
 import { Logger } from "./logger";
-import { AnalyticsUtils } from "./utils/analytics";
-
-export function isVSCodeTelemetryEnabled(): boolean | undefined {
-  // `isTelemetryEnabled` only seems to be available on VSCode 1.55 and above.
-  // Our dependencies are currently set up for 1.54, so typescript does not understand that it exists.
-  try {
-    return (vscode.env as any).isTelemetryEnabled;
-  } catch (err) {
-    Logger.error({ msg: "Error when trying to access telemetry preference" });
-    return undefined;
-  }
-}
 
 /** Creates a SegmentClient for telemetry, if enabled, and listens for vscode telemetry settings to disable it when requested. */
 export function setupSegmentClient(ws: DWorkspaceV2, cachePath?: string) {
   function instantiateSegmentClient() {
-    // if the current status was set by configuration, and that configuration has changed, we should update it and report the change
-    const status = SegmentClient.getStatus();
-    if (SegmentClient.setByConfig(status)) {
-      const disableTelemetry = ConfigUtils.getWorkspace(
-        ws.config
-      ).disableTelemetry;
-      if (
-        SegmentClient.isDisabled(status) &&
-        !disableTelemetry &&
-        isVSCodeTelemetryEnabled()
-      ) {
-        // was disabled, now enabled
-        Logger.info({
-          msg: "The user changed VSCode or workspace settings, so we are now disabling telemetry",
-        });
-        const reason = TelemetryStatus.ENABLED_BY_CONFIG;
-        SegmentClient.enable(reason);
-        AnalyticsUtils.track(VSCodeEvents.EnableTelemetry, { reason });
-      } else if (SegmentClient.isEnabled(status)) {
-        // was enabled, now disabled
-        if (disableTelemetry) {
-          Logger.info({
-            msg: "The user change workspace settings, so we are now disabling telemetry",
-          });
-          const reason = TelemetryStatus.DISABLED_BY_WS_CONFIG;
-          AnalyticsUtils.track(VSCodeEvents.DisableTelemetry, { reason });
-          SegmentClient.disable(reason);
-        } else if (!isVSCodeTelemetryEnabled()) {
-          Logger.info({
-            msg: "The user change VSCode settings, so we are now disabling telemetry",
-          });
-          const reason = TelemetryStatus.DISABLED_BY_VSCODE_CONFIG;
-          AnalyticsUtils.track(VSCodeEvents.DisableTelemetry, { reason });
-          SegmentClient.disable(reason);
-        }
-      }
-    }
-
+    const disabledByWorkspace = ConfigUtils.getWorkspace(
+      ws.config
+    ).disableTelemetry;
     const segment = SegmentClient.instance({
       forceNew: true,
       cachePath,
+      disabledByWorkspace,
```

Comment:
Are we now ignoring the VSCode's "is telemetry enabled" setting?

Reply:
yes. the way we currently detect for and try to update telemetry setting based on vscode settings is rather complicated. we might re-introduce in the future